template/registry/registry.go

```diff
@@ -0,0 +1,153 @@
+package registry
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+)
+
+type Registry []Func
```

Comment:
Why not `map[string]Func`? It speeds up Func lookup.

Reply:
dunno :)